### ISSUE TITLE
Newly created person shows in LHS

### ISSUE THREAD

**Issue by abbyad** [medic/medic-webapp#2841]
Logged in as admin, you should not see any people in the LHS. When adding a person they show in the LHS, until you reload.

Added "Zorba", shown in LHS
![image](https://cloud.githubusercontent.com/assets/789512/19455859/c567a592-948c-11e6-9d34-5df2b7142ba6.png)

Proper LHS only after reload:
![image](https://cloud.githubusercontent.com/assets/789512/19455903/f37afc18-948c-11e6-9be3-e9bd5191c0d6.png)

Running 2.9.0-alpha.4331

**Comment by sglangevin** [comment]
I've seen this happen on start-up (first view of People tab) as well. I logged in and saw 1 person in the LHS. The person disappeared after a refresh and didn't show up in the LHS the next time I logged in.

**Comment by garethbowen** [comment]
This would also happen if a person was replicated into the database while you had the People screen open. The problem is we're watching for changes in the database and then updating the list when a change comes in. This worked fine until we filtered the LHS more heavily. I think the fix is to fire off a background search when changes occur so the current filters are used, rather than blinding appending the new contact to the list.

**Comment by sglangevin** [comment]
@garethbowen does this also fix the fact that new reports show up in History that don't fit the filter that you have put on the History tab? For example, I filtered to see CHW Area A's reports and while I was looking at them, reports from CHW Area B came in and appeared in the list. If not, I'll file a separate issue for that.

**Comment by garethbowen** [comment]
Yeah that's the same root cause and I'll apply the same fix to both reports and contacts.

**Comment by garethbowen** [comment]
PR: https://github.com/medic/medic-webapp/pull/2867

The Changes listener used to always append the new item to the list which meant we ignored any filters that were set. The listeners are now in the controller and just fire off a background search using the current filters.

**Comment by SCdF** [comment]
So this code looks good @garethbowen, and feel free to merge it, but may I ask: do we actually want the search to magically update in the background because a new doc was added? Is the chattiness worth the functionality?

**Comment by garethbowen** [comment]
I've split off #2870 to alleviate some performance concerns and will hopefully get that done this iteration too.
